src/stubs/ES6Map.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ES6Map
+ */
+
+'use strict';
+
+function cloneWithoutKey(obj, notKey) {
+  var newObj = {};
+  var keys = Object.keys(obj);
+  for (var i = 0, l = keys.length; i < l; i++) {
+    var key = keys[i];
+    if (key !== notKey) {
+      newObj[key] = obj[key];
+    }
+  }
+  return newObj;
+}
+
+if (typeof Map !== 'undefined') {
```

Comment:
I'm suuuuuuuuuper weary of this, there are tons of O(n) polyfills out there and it would be a bad day if you were using one. One solution is to check for `[native code]` in `Map.toString()`.